app/src/main/java/com/hedvig/app/feature/language/LanguageAndMarketViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package com.hedvig.app.feature.language
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.content.Intent
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import androidx.preference.PreferenceManager
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.feature.marketpicker.Market
+import com.hedvig.app.feature.marketpicker.MarketModel
+import com.hedvig.app.feature.marketpicker.MarketRepository
+import com.hedvig.app.feature.settings.Language
+import com.hedvig.app.feature.settings.LanguageModel
+import com.hedvig.app.feature.settings.SettingsActivity
+import com.hedvig.app.makeLocaleString
+import kotlinx.coroutines.launch
+
+class LanguageAndMarketViewModel(
+    private val languageRepository: LanguageRepository,
+    private val marketRepository: MarketRepository,
+    application: Application
+) : AndroidViewModel(application) {
```

Comment:
```suggestion
) : ViewModel() {
```
If you need to access the Application Context, you can inject it via the constructor. The Context provided by dependency injection is always the Application Context.